### ISSUE TITLE
Failing assertion message does not make sense with DateTime

### ISSUE THREAD

**Issue by hiredgunhouse** [fluentassertions/fluentassertions#694]
I have a test where comparing two dates results in a failed assertion with following message:
```
Xunit.Sdk.XunitException: Expected <2017-11-23 16:10:54.350>, but found <2017-11-23 16:10:54.350>.
   at FluentAssertions.Execution.XUnit2TestFramework.Throw(String message) in 
```
Dates do not match after one of them does a round trip to and back from the DB and they most certainly differ, but I assume the assertion message contains to low precision to make it visible.
A little brain fuck at first, then just a minor annoyance, you decide whether that's something you want to "fix".
BTW. I can create a repo project if you want.

**Comment by jnyrup** [comment]
Thanks for pointing it out.
It's an annoyance when the failure message doesn't reveal the error (especially when bragging about clear failure messages).

The resolution when formatting of DateTime was improved a lot in [5.0.0.0-beta4](https://github.com/fluentassertions/fluentassertions/releases/tag/5.0.0-beta.4).

The results when using 5.0.0-beta4:

```c#
DateTime dateTime = new DateTime(2016, 06, 04).AddTicks(1);
DateTime otherDateTime = dateTime.AddTicks(1);

dateTime.Should().Be(otherDateTime);
```

now fails with 
> Message: Expected date and time to be <2016-06-04 00:00:00.0000002>, but found <2016-06-04 00:00:00.0000001>.

**Comment by dennisdoomen** [comment]
@jnyrup Isn't this now fixed with #695

**Comment by jnyrup** [comment]
@dennisdoomen the issue described here should already have been fixed with #669.
#695 handles the special case when hour=0, minute=0 and second=0, but milliseconds, microsecond or nanosecond is not 0.

**Comment by dennisdoomen** [comment]
Should be fixed in beta 5 then, right?

**Comment by jnyrup** [comment]
Beta 4 fixed most cases with #669 
Beta 5 should (hopefully) have fixed all cases with #695.

The posted failure message indicates that a version <5.0.0-beta4 was used, since it doesn't include microseconds or nanoseconds.

**Comment by dennisdoomen** [comment]
OK, then we'll close it.
